### PR TITLE
[SOL-1720] Fix incorrect info in coupon report

### PR DESCRIPTION
**Pull request by marjev**
@mjfrey Here's a PR for [SOL-1720](https://openedx.atlassian.net/browse/SOL-1720). Please take a look.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @marjev! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?number=645&repo=edx%2Fecommerce

File: ecommerce/extensions/voucher/utils.py [hunk 1]
```diff
@@ -67,6 +70,11 @@ def generate_coupon_report(coupon_vouchers):
     for coupon_voucher in coupon_vouchers:
         coupon = coupon_voucher.coupon
 
+        # Get client based on the invoice that was created during coupon creation
+        basket = Basket.objects.filter(lines__product_id=coupon.id).first()
+        order = Order.objects.get(basket=basket)
+        invoice = Invoice.objects.get(order=order)
```
Comment (mjfrey) [inline ecommerce/extensions/voucher/utils.py]: @marjev need to catch an exception here as older coupons may not have an order in the invoice.  So please account for older and newer invoices.  Some have baskets and some will have orders.

File: ecommerce/extensions/voucher/utils.py [hunk 2]
```diff
@@ -116,8 +127,8 @@ def generate_coupon_report(coupon_vouchers):
                 'Coupon Type': coupon_type,
                 'URL': URL,
                 'CourseID': course_id,
-                'Partner': seat_stockrecord.partner.name,
-                'Client': Basket.objects.filter(lines__product_id=coupon.id).first().owner.username,
+                'Partner': course_organization,
+                'Client': invoice.client.name,
```
Comment (mjfrey) [inline ecommerce/extensions/voucher/utils.py]: There also may not be a client in an invoice as older coupons did not have this.

File: ecommerce/extensions/voucher/utils.py (hunk 1 above)
Comment (marjev) [inline ecommerce/extensions/voucher/utils.py]: Done.

File: ecommerce/extensions/voucher/utils.py (hunk 2 above)
Comment (marjev) [inline ecommerce/extensions/voucher/utils.py]: Done.

**Comment by mjfrey** [comment]
:+1: 
Please squash.

**Pull request merged**